test/behavior/asm.zig

```diff
@@ -149,11 +149,11 @@ test "struct/array/union types as input values" {
     ); // fails
     asm volatile (""
         :
-        : [_] "m" (@as(struct { x: u32, y: u8 }, undefined)),
+        : [_] "m" (@as(packed struct { x: u32, y: u8 }, undefined)),
     ); // fails
     asm volatile (""
         :
-        : [_] "m" (@as(union { x: u32, y: u8 }, undefined)),
+        : [_] "m" (@as(packed union { x: u32, y: u8 }, undefined)),
     ); // fails
```

Comment:
Maybe add some tests here for `extern` aggregates too, since that can also work okay?